src/components/marketplace/order_details.tsx

```diff
@@ -70,18 +90,120 @@ class OrderDetails extends React.Component<Props, State> {
+                const totalCostInWeth = totalPriceWithoutFeeInWeth.add(zeroXFeeInWeth);
+
+                /* Calculates total cost in USD */
+                const zeroXFeeInUSD = zeroXPriceInUSD.mul(MAKER_FEE);
+                const totalPriceWithoutFeeInUSD = totalPriceWithoutFeeInWeth.mul(ethInUSD);
+                const totalCostInUSD = totalPriceWithoutFeeInUSD.add(zeroXFeeInUSD);
+
+                this.setState({
+                    limitOrder: {
+                        ...this.state.limitOrder,
+                        zeroXFeeInWeth,
+                        zeroXFeeInUSD,
+                        totalCostInWeth,
+                        totalCostInUSD,
+                    },
+                });
+            } catch (error) {
+                throw error;
+            }
+        }
+    };
+
+    public updateMarketOrderState = async () => {
+        /* TODO */
+        const { tokenAmount, tokenPrice } = this.props;
+        await this._calculateTotalCostMarketInWeth(tokenAmount, tokenPrice);
+        return null;
+    };
+
+    public componentDidUpdate = async (prevProps: Readonly<Props>, prevState: Readonly<State>, snapshot?: any) => {
+        const newProps = this.props;
+        if (
+            newProps.tokenPrice !== prevProps.tokenPrice ||
+            newProps.orderType !== prevProps.orderType ||
+            newProps.tokenAmount !== prevProps.tokenAmount ||
+            newProps.selectedToken !== prevProps.selectedToken
+        ) {
+            if (newProps.orderType === OrderType.Limit) {
+                await this.updateLimitOrderState();
+            }
+            if (newProps.orderType === OrderType.Market) {
+                await this.updateMarketOrderState();
+            }
+        }
     };
 
     public componentDidMount = async () => {
-        let zeroXPriceInUSD = await getZeroXPriceInUSD();
-        zeroXPriceInUSD = zeroXPriceInUSD.mul(MAKER_FEE);
-        this.setState({ zeroXPriceInUSD });
+        const { orderType } = this.props;
+        if (orderType === OrderType.Limit) {
+            await this.updateLimitOrderState();
+        }
+        if (orderType === OrderType.Market) {
+            await this.updateMarketOrderState();
+        }
     };
 
     public render = () => {
-        const { orderDetailType } = this.state;
+        const { orderType } = this.props;
+        let render = null;
+        if (orderType === OrderType.Limit) {
+            const { orderDetailType, zeroXFeeInUSD, totalCostInWeth, totalCostInUSD } = this.state.limitOrder;
+            render = this._renderOrderDetails(orderDetailType, zeroXFeeInUSD, totalCostInWeth, totalCostInUSD);
+        }
+        if (orderType === OrderType.Market) {
+            const { orderDetailType, zeroXFeeInUSD, totalCostInWeth, totalCostInUSD } = this.state.marketOrder;
+            render = this._renderOrderDetails(orderDetailType, zeroXFeeInUSD, totalCostInWeth, totalCostInUSD);
```

Comment:
Req: This line and line 183 are equal, they could be outside the `if`s.

Reply:
I'd just changed it